src/graphql_relay/connection/arrayconnection.py

```diff
@@ -1,6 +1,7 @@
 import binascii
 import warnings
-from typing import Any, Optional, Sequence
+from typing import Any, Iterator, Optional, Sequence
+from typing_extensions import Protocol
```

Comment:
You try importing from typing and only if that fails with an ImportError, import from typing_extensions.